### PR TITLE
Fix jms-cluster build

### PR DESCRIPTION
**Pull request by groldan**
Fixes the build for the `jms-cluster` module whose tests were failing

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Committs changing the REST API, or any configuration object, should check it the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by aaime** [comment]
I have very little experience in this module, just "dumped" it here years ago, the original author was a colleague that is no longer working with us. @nmco is the closest person to a maintainer I believe.

Aside, I see a large diff and a ton of reformat like changes, are they necessary?

File: src/community/jms-cluster/jms-commons/src/main/java/org/geoserver/cluster/client/JMSQueueListener.java [hunk 1]
```diff
@@ -118,53 +105,48 @@ public void onMessage(Message message, Session session) throws JMSException {
...
-                            "Unable to synchronize message locally.\n SPI: " + generatorClass);
-                }
-
-            } catch (Exception e) {
-                final JMSException jmsE = new JMSException(e.getLocalizedMessage());
-                jmsE.initCause(e);
-                throw jmsE;
-            } finally {
-                this.consumedEvents.incrementAndGet();
+        final ObjectMessage objMessage = (ObjectMessage) (message);
```
Comment (nmco) [inline src/community/jms-cluster/jms-commons/src/main/java/org/geoserver/cluster/client/JMSQueueListener.java]: So basically here we always assume that indeed we get an ``ObjectMessage``, wondering any specific reason to get ride of the origin ``if``?

File: src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/CatalogUtils.java [hunk 2]
```diff
@@ -6,14 +6,18 @@
 package org.geoserver.cluster.impl.handlers.catalog;
```
Comment (nmco) [inline src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/CatalogUtils.java]: Not sure what happen here, was the class reformat, is there any actual code change?

File: src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/JMSCatalogAddEventHandler.java [hunk 3]
```diff
@@ -63,6 +64,7 @@ public boolean synchronize(CatalogEvent event) throws Exception {
             }
 
         } catch (Exception e) {
+            e.printStackTrace();
```
Comment (nmco) [inline src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/JMSCatalogAddEventHandler.java]: A forgotten quick test?

**Comment by nmco** [comment]
Any news on this one @groldan? Looks like the conflicts come from the mass cleaning refactoring done a few weeks ago.

That say, I spend a bit more of time today looking at this pull request. IMHO this PR as it is is too complex, the usual process will need to be followed here :
- Creating a JIRA ticket for the identified issue(s) with the test(s), and produce focused PRs for them.
- Separated JIRA and PRs for the refactoring, I see several types of refactoring: logging, code, exception handling ...

**Comment by aaime** [comment]
@nmco asked for clarifications and a better restructuring of this PR on the list and here, did not see a follow up.
Been a couple of months, shall we close the PR?

**Comment by groldan** [comment]
> @nmco asked for clarifications and a better restructuring of this PR on the list and here, did not see a follow up.
> Been a couple of months, shall we close the PR?

I'll try to find the time to re-work this PR asap, on volunteer basis though so it might need to wait for some bored weekend, meanwhile please don't close.

**Comment by aaime** [comment]
@groldan last update on this one is a few months old, should we still keep it open?

**Comment by aaime** [comment]
Closing due to lack of feedback

**Comment by groldan** [comment]
Ok. I've spent 8hs last night trying to adapt the patch to not include all the refactorings but only the fixes. Turns out I couldn't make it work. It's funny you complained this module is being used in production but it doesn't build at least since last year, and sad I couldn't provide a patch that was acceptable by the module maintainers.

All in all it's hard _for me_ to work on this piece of code becase the tests just expect a number of seemingly arbitrary events without enough granularity as to know what's expected, and general code quality issues that make it hard to follow.

IMHO this pr, if you looked at it commit by commit, they make a lot of sens and clean up the code meanwhile. Sorry I failed to understand the nature of the failures without those refactorings, some where indeed needed, like CatalogUtils not always having to create a new object and hence being split into "creating" and "checking" singletons.

Anyhow, here's a branch with a bunch of commits I think would be of value: https://github.com/groldan/geoserver/tree/fix_jms_cluster_tests/master

**Comment by aaime** [comment]
@groldan it is indeed still used in some cases. A community module does not need to pass tests.

@nmco asked you to propose a PR that was easier to understand, the template is also clear about how the review process is done: "Reviewing is a process done by project maintainers, mostly on a volunteer basis"
We cannot dedicate a large amount of time to reviews, so PRs should be first and foremost easy to understand.

If you can split the PR in smaller chunks, or negotiate with Nuno a different way to handle it (a call? comments in the PR? up to you, I have no dog in this fight). 

I closed because in May you said you'd rework the PR "ASAP", months have passed with no further feedback, last week I asked again, and got no answer. That's all. Just tell me you actually want to work on it, and I'll re-open it.

File: src/community/jms-cluster/jms-commons/src/main/java/org/geoserver/cluster/client/JMSQueueListener.java (hunk 1 above)
Comment (groldan) [reply src/community/jms-cluster/jms-commons/src/main/java/org/geoserver/cluster/client/JMSQueueListener.java:115]: We do, the factory used to create them creates only `ObjectMessage`s

Comment (groldan) [reply src/community/jms-cluster/jms-commons/src/main/java/org/geoserver/cluster/client/JMSQueueListener.java:115]: In fact, `JMSQueueListener` should implement `SessionAwareMessageListener<ObjectMessage>` not `SessionAwareMessageListener<Message>`

**Comment by groldan** [comment]
yikes, I'm just sorry and sad I couldn't find the time to get this done, you're right on the timing issues. I currently have no use for this module, yet I think it's a waste that both the fixes (adds support for WMTSStore/Info, fixes some issues with styles sync'ing) and the code clean up would be lost. May I find the time to provide a separate pr, I'll do.
@nmco if, on the other hand, you're working on this module and can take a while to look at the patch as a whole, there are things that could be of interest.

As a separate note, one thing I found that could be a problem, is that nodes sync' up resources upon pre-update events, when IMHO they should do so upon post-update events. Say something goes wrong in the event-sender side of the fence between pre-update and post-update, and you'll end up with an inconsistent state.
